tests/test_downloadermiddleware_retry.py

```diff
@@ -110,19 +112,49 @@ def test_twistederrors(self):
             == 2
         )
 
-    def _test_retry_exception(self, req, exception):
+    def test_exception_to_retry_added(self):
+        exc = ValueError
+        self.mw.exceptions_to_retry = self.mw.exceptions_to_retry + (exc,)
```

Comment:
I suspect this line is pointless and can be removed completely. And maybe `ValueError` can be replaced with `exc` in the line below, to avoid repeating `ValueError`.